server/Main.hs

```diff
@@ -157,6 +181,35 @@ server externs initNamesEnv initEnv port = do
           Scotty.json $ A.object [ "error" .= err ]
         Right (warnings, comp) ->
           Scotty.json $ A.object [ "js" .= comp, "warnings" .= warnings ]
+    
+    get "/complete" $ do
+      query <- param "q"
+      Scotty.setHeader "Access-Control-Allow-Origin" "*"
+      Scotty.setHeader "Content-Type" "application/json"
+      let ideClient = 
+            Process.createProcess_ "purs-ide-client"
+              (Process.proc "purs" ["ide", "client"])
+                { Process.std_in = Process.CreatePipe
+                , Process.std_out = Process.CreatePipe
+                }
+          mkCommand q = A.encode $ A.object
+                        [ "command" .= ("complete" :: Text)
+                        , "params" .= A.object 
+                          [ "filters" .= A.Array
+                            ( V.fromList 
+                              [ A.object
+                                [ "filter" .= ("prefix" :: Text) 
+                                , "params" .= A.object
+                                  [ "search" .= q ]
+                                ]
+                              ]
+                            )
+                          ]
+                        ]
+      (Just handleIn, Just handleOut, _, _) <- liftIO ideClient
+      liftIO $ Char8.hPutStrLn handleIn (mkCommand (query :: Text))
+      result <- liftIO $ BS.hGetContents handleOut
+      Scotty.text (TL.fromStrict (T.decodeUtf8 result))
```

Comment:
This currently works, but there might be some inefficiency here that I'm not spotting. Running `purs ide client` directly (talking to same `purs ide server` instance) is instantaneous, while hitting this endpoint takes around 2s, sometimes longer.

![2022-02-06 16 00 04](https://user-images.githubusercontent.com/26548438/152701156-63c1ea94-f0f3-4b85-bd24-6e2e5f02833b.gif)

Reply:
Maybe this could be a POST request with JSON constructed on the client, and passed directly to `purs ide client`

Reply:
It's somehow much faster using POST https://github.com/purescript/trypurescript/pull/270/commits/749670093a22c5b2abaee654a9e2051be302fee1, but I don't know if there's any security issues with just proxying directly to `purs ide server`